sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Config/ServiceBusClientFactory.cs

```diff
@@ -0,0 +1,226 @@
+            _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
+        }
+
+        public virtual MessageProcessor CreateMessageProcessor(string entityPath, string connection)
+        {
+            return new MessageProcessor(GetOrAddProcessor(entityPath, connection));
+        }
+
+        public ServiceBusProcessor CreateProcessor(string entityPath, string connection)
+        {
+            return GetOrAddProcessor(entityPath, connection);
+        }
+
+        public ServiceBusSender CreateMessageSender(string entityPath, string connection)
+        {
+            Argument.AssertNotNullOrEmpty(entityPath, nameof(entityPath));
+
+            return GetOrAddMessageSender(entityPath, connection);
+        }
+
+        public virtual ServiceBusReceiver CreateBatchMessageReceiver(string entityPath, string connection)
+        {
+            Argument.AssertNotNullOrEmpty(entityPath, nameof(entityPath));
+
+            return GetOrAddMessageReceiver(entityPath, connection);
+        }
+
+        public SessionMessageProcessor CreateSessionMessageProcessor(string entityPath, string connection)
+        {
+            Argument.AssertNotNullOrEmpty(entityPath, nameof(entityPath));
+
+            return new SessionMessageProcessor(GetOrAddSessionProcessor(entityPath, connection));
+        }
+
+        public ServiceBusAdministrationClient CreateAdministrationClient(string entityPath, string connection)
+        {
+            Argument.AssertNotNullOrEmpty(entityPath, nameof(entityPath));
+            var connectionInfo = ResolveConnectionInformation(connection);
+            if (connectionInfo.ConnectionString != null)
+            {
+                return new ServiceBusAdministrationClient(connectionInfo.ConnectionString);
+            }
+            else
+            {
+                return new ServiceBusAdministrationClient(connectionInfo.FullyQualifiedNamespace, connectionInfo.Credential);
+            }
+        }
+
+        private ServiceBusSender GetOrAddMessageSender(string entityPath, string connection)
+        {
+            ServiceBusClient client = GetOrCreateClient(connection);
+            return _messageSenderCache.GetOrAdd(entityPath, client.CreateSender(entityPath));
+        }
+
+        private ServiceBusReceiver GetOrAddMessageReceiver(string entityPath, string connection)
+        {
+            ServiceBusClient client = GetOrCreateClient(connection);
+
+            return _messageReceiverCache.GetOrAdd(entityPath, (_) => client.CreateReceiver(
+                entityPath,
+                new ServiceBusReceiverOptions
+            {
+                PrefetchCount = _options.PrefetchCount
+            }));
+        }
+
+        private ServiceBusProcessor GetOrAddProcessor(string entityPath, string connection)
+        {
+            ServiceBusClient client = GetOrCreateClient(connection);
+
+            return _processorCache.GetOrAdd(entityPath, (_) =>
```

Comment:
Can different connection have the same `entityPath`?

Reply:
yes they can